src/components/tl/cuda/tl_cuda_coll.h

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_TL_CUDA_COLL_H_
+#define UCC_TL_CUDA_COLL_H_
+
+#include "tl_cuda.h"
+
+#define TASK_TEAM(_task)                                                       \
+    (ucc_derived_of((_task)->super.team, ucc_tl_cuda_team_t))
+
+#define TASK_ARGS(_task) (_task)->super.bargs.args
+
+#define TASK_SYNC(_task, _rank)                                                \
+    ({                                                                         \
+        ucc_tl_cuda_team_t *_team = TASK_TEAM(_task);                          \
+        UCC_TL_CUDA_TEAM_SYNC(_team, _rank, (_task)->coll_id);                 \
+    })
+
+static inline void ucc_tl_cuda_task_reset(ucc_tl_cuda_task_t *task)
+{
+    task->super.super.status = UCC_INPROGRESS;
```

Comment:
"reset" is tricky imho. Originally "reset" was added to support persistent behavior or frags restart in fragmented schedules (with pipeline depth < num_of frags). Since task has a fixed seq_num during its lifetime, after "reset" you will have all remote shm sync values equal to that seq_num from the previous execution. THis will imply sync will be incorrect.

I guess we should do task->seq_num++ each reset, while task->coll_id will be fixed to use same sync each time. wdyt?

Reply:
fixed